source/MBF.SAMD51.GPIO.pas

```diff
@@ -42,14 +42,37 @@ interface
   Pad2=$120000;
   Pad3=$130000;
 
+  PINMODE_INPUT           = 0;
+  PINMODE_OUTPUT          = 1;
+  PINMODE_INPUT_PULLUP    = 2;
+  PINMODE_INPUT_PULLDOWN  = 3;
+
+
 type
   TPinLevel=(Low=0,High=1);
   TPinValue=0..1;
   TPinIdentifier=-1..160;
-  TPinMode = (Input=%00, Output=%01, Analog=%11, AF0=$10, AF1, AF2, AF3, AF4, AF5, AF6, AF7, AF8, AF9, AF10, AF11, AF12, AF13);
+
+  TPinMode = (Input=%00, Output=%01);
   TPinDrive = (None=%00,PullUp=%01,PullDown=%10);
   TPinOutputMode = (PushPull=0,OpenDrain=1);
   TPinOutputStrength = (Normal=%0, High=%1);
+
+  eAnalogReference = (
```

Comment:
Why the e Prefix, what does it mean error?

Reply:
e is the c-prefic for an enumeration. As b is for boolean ....
Its just a 1 to 1 copy of the c-enumeration. So, name may be changed to whatever we find suitable.